### DISCUSSION TITLE
File scoped usings?

### DISCUSSION THREAD

**Discussion by virzak** [CommunityToolkit/dotnet#794] [Q&A] [answered]
Why doesn't generated code declare namespaces and instead writes out 
```cs
private global::CommunityToolkit.Mvvm.Input.RelayCommand? ...
```
?

Looks like source generator are about split in half on the approach.

This question came up here: zompinc/sync-method-generator#13

**Comment by Sergio0694** [comment, marked as answer]
I've written a detailed explanation in a comment in the runtime repo, let me go pick that up and also share it there 🙂

**EDIT:** see https://github.com/dotnet/runtime/pull/94428#issuecomment-1796294871 and comments below.
